algorithms/src/sorting/Kokkos_SortPublicAPI.hpp

```diff
@@ -219,6 +258,10 @@ std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value> sort(
 
 template <class ViewType>
 void sort(ViewType view, size_t const begin, size_t const end) {
+  // same constraints as the overload above which this gets dispatched to
+  static_assert(ViewType::rank == 1,
```

Comment:
These all need to call the static member function `rank()` if `DynRankView` of rank 1 is to be supported.

Reply:
this won't work for binsort beause of this: https://github.com/kokkos/kokkos/blob/develop/algorithms/src/sorting/impl/Kokkos_CopyOpsForBinSortImpl.hpp